pkg/controllers/node/emptiness.go

```diff
@@ -65,7 +64,7 @@ func (r *Emptiness) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisi
 		return reconcile.Result{}, nil
 	}
 
-	emptinessTimestamp, hasEmptinessTimestamp := n.Annotations[v1alpha5.EmptinessTimestampAnnotationKey]
+	_, hasEmptinessTimestamp := n.Annotations[v1alpha5.EmptinessTimestampAnnotationKey]
 	if !empty {
```

Comment:
I think this whole thing can be compressed into

```go
if !empty && hasEmptinessTimestamp {
  delete(n.Annotations, v1alpha5.EmptinessTimestampAnnotationKey)
  logging.FromContext(ctx).Infof("Removed emptiness TTL from node")
} else if empty && !hasEmptinessTimeStamp {
  n.Annotations[v1alpha5.EmptinessTimestampAnnotationKey] = r.clock.Now().Format(time.RFC3339)
  logging.FromContext(ctx).Infof("Added TTL to empty node")
}
return reconcile.Result{}, nil
```

Reply:
Good idea! Let me condense it.